eclipse-theia/theia
# added task id to task config interface

- "Run task" and "Run task group" use task label, source folder, and scope to "guess" which task is the one that users want to run.
  With this change, the task identification is added to the task config interface, and it is used to search tasks.
- Reset "Last task" when "Task: Clear history" is executed.
- "Run last task" uses task id to search the most recent task, if not found, use the task type, label, and source to perform one more search.
- With the introduction of task id, Theia supports defining configured tasks that have same labels. However, to comply with VS Code, if a contributed task is customized more than once, the last customization overwrites the prior ones.
- make sure $fetchTasks() does not return duplicated contributed tasks that are customized.
- added extension / plugin id to the TaskDefintion interface.



Signed-off-by: Liang Huang <lhuang4@ualberta.ca>

#### How to test

- Theia should support the definiton of multiple tasks that have the same label
- Theia should NOT support having multiple customizations for the same contributed task. If there are more than one customizations, the last customization overwrites the priors.
- "Last task" should be reset after "Task: Clear history" is executed.

Smoke test the followings:
- Run a task
- Configure a configured / contributed task
- Run last task
- Run build / test task

#### Review checklist

- [x] Approval of CQ https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22251
